### PR TITLE
ps: support the `container...` notation for `ps --filter network=...`

### PR DESCRIPTION
**Pull request by flouthoc**
Hi team,

Following PR extends `ps --filter` to support docker like:
 `podman container ps --filter network=container:<CtrIdorCtrName>`.

Closes: https://github.com/containers/podman/issues/10361

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon @Luap99 PTAL

File: libpod/container.go
Comment (Luap99) [inline libpod/container.go:1178]: I think you copied the logic from inspect. It would make sense to use the complete inspect logic here and use this function in inspect to prevent code duplication.

File: pkg/domain/filters/containers.go
Comment (Luap99) [inline pkg/domain/filters/containers.go:214]: Can you move this down to line 233.

File: pkg/domain/filters/containers.go
Comment (Luap99) [inline pkg/domain/filters/containers.go:226]: Please handle the error.

File: pkg/domain/filters/containers.go
Comment (Luap99) [inline pkg/domain/filters/containers.go:231]: You can return false here, if the network mode is `container:` we do not have to check network names.

File: test/e2e/ps_test.go
Comment (Luap99) [inline test/e2e/ps_test.go:186]: Can you add another test with `container:<ID>` instead of name.

File: libpod/container.go
Comment (flouthoc) [reply libpod/container.go]: @Luap99 thanks this is resolved in latest commit.

File: pkg/domain/filters/containers.go
Comment (flouthoc) [reply pkg/domain/filters/containers.go]: Resolved in latest commit.

File: test/e2e/ps_test.go
Comment (flouthoc) [reply test/e2e/ps_test.go:186]: Resolved in latest commit.

File: pkg/domain/filters/containers.go
Comment (flouthoc) [reply pkg/domain/filters/containers.go]: resolved :)

File: pkg/domain/filters/containers.go
Comment (flouthoc) [reply pkg/domain/filters/containers.go:236]: resolved in latest commit.

**Review by Luap99 — APPROVED** [review]
LGTM

File: libpod/container.go
Comment (mheon) [inline libpod/container.go:1178]: Why does this accept a spec? The container struct already contains a spec, and nothing you're accessing here requires something that would be overwritten from the initial user-provided spec. Just using `c.config.Spec` is simpler and safer.

File: pkg/domain/filters/containers.go
Comment (mheon) [inline pkg/domain/filters/containers.go:222]: Why `strings.Contains()` instead of `strings.HasPrefix()`?

File: pkg/domain/filters/containers.go
Comment (mheon) [inline pkg/domain/filters/containers.go:223]: Probably safer to do a length check here to ensure we don't panic.

File: pkg/domain/filters/containers.go
Comment (mheon) [reply pkg/domain/filters/containers.go:222]: As I look at this more - it really looks as though the check for `container:` needs to be integrated into the `filterValues` loop below, instead of done separately up here.

File: libpod/container.go
Comment (flouthoc) [reply libpod/container.go:1181]: @mheon Following function is also going to be used by `generateInspectContainerHostConfig` in inspect code chunk. I think for both `inspect` and `ps --filter` their exists a state where config does not exist **(e.g., because the container was never started)** for such cases I think we can read data from db.

However for inspect block `ctrSpec` is already fetched from state so i thought would be more optimal to reuse the `ctrSpec` as argument inoder to reduce lookup cost.

File: pkg/domain/filters/containers.go
Comment (flouthoc) [reply pkg/domain/filters/containers.go]: This is resolved in latest commit.

File: pkg/domain/filters/containers.go
Comment (flouthoc) [reply pkg/domain/filters/containers.go]: This is resolved in latest commit.

**Comment by flouthoc** [comment]
@mheon I have resolved all your feedback comments in latest commit. I have a doubt on first feedback , could you please take a look and correct me if i am wrong somewhere.

**Comment by flouthoc** [comment]
Re-basing with main to add fix for https://github.com/containers/podman/pull/11077

File: libpod/container.go
Comment (mheon) [reply libpod/container.go:1180]: `c.config.Spec` is always valid; there is never a point at which it is not populated. It is not the final spec that is used for the container (we use it to generate the final spec), but it is very close - the changes made to produce a final spec are minimal. As such it is safe to use here - no part of what you are accessing will change during final spec generation.

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons
I'll let @mheon press the final merge button as I'm not certain all of his concerns were addressed.

**Comment by flouthoc** [comment]
@TomSweeneyRedHat thanks :) there is one change left which @mheon suggested. @mheon made relevant changes will only try to read spec from db if `c.config.Spec` is nil for added safety does it looks fine to you ?

File: libpod/container.go
Comment (mheon) [inline libpod/container.go:1184]: This will never be true, you can drop the check - we will never create a container with a nil state.

Comment (mheon) [reply libpod/container.go:1184]: Further - we create the spec that `specFromState()` would retrieve *from* `c.config.Spec` so that will also fail if `c.config.Spec` does not exist

Comment (flouthoc) [reply libpod/container.go]: @mheon thanks, I have removed the check in latest commit.

**Comment by flouthoc** [comment]
Just a sec cleaning error handling for removed block.

**Comment by flouthoc** [comment]
@mheon This is resolved could you please review again.

**Pull request merged**
